subsys/bluetooth/host/l2cap.c

```diff
@@ -2991,6 +3002,31 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 		return -EBUSY;
 	}
 
+	/* default (max) MPS for channels */
+	if (mps == 0) {
+		mps = BT_L2CAP_RX_MTU;
+	}
+
+	/* MPS shall not be bigger than MTU + BT_L2CAP_SDU_HDR_SIZE
+	 * as the remaining bytes cannot be used.
+	 */
+	mps = MIN(mtu + BT_L2CAP_SDU_HDR_SIZE, mps);
+
+	for (int j = 0; j < i; j++) {
+		ch = BT_L2CAP_LE_CHAN(chans[j]);
+		/* For now simply don't allow MPS changes on channels using
+		 * segment direct API
```

Comment:
Is this a behavior change in the stack? 

If so, do we need to update the release notes/migration guide? 
Is this behavior documented in the public API?

Reply:
This public API `bt_l2cap_ecred_chan_reconfigure` does not take an `mps` parameter at all, so there is no good way for the application to provide a new MPS value.

Reply:
this is experimental API and reconfigure is currently not working well for it anyway (is application uses MPS<BT_L2CAP_RX_MTU)